arrows/ocv/resection_camera.cxx

```diff
@@ -0,0 +1,218 @@
+
+/// \file
+/// \brief OCV resection_camera algorithm implementation
+
+#include "camera_intrinsics.h"
+#include "resection_camera.h"
+
+#include <opencv2/calib3d/calib3d.hpp>
+#include <opencv2/core/eigen.hpp>
+
+using namespace std;
+using namespace cv;
+
+namespace kwiver {
+
+namespace arrows {
+
+namespace ocv {
+
+// ----------------------------------------------------------------------------
+
+struct resection_camera::priv
+{
+  double reproj_accuracy = 4.;
+  int max_iterations = 32;
+  vital::logger_handle_t m_logger;
+
+  priv() : m_logger{ vital::get_logger( "arrows.ocv.resection_camera" ) }
+  {
+  }
+};
+
+// ----------------------------------------------------------------------------
+
+resection_camera
+::resection_camera()
+  : d_{ new priv }
+{
+}
+
+resection_camera::~resection_camera()
+{
+}
+
+// ----------------------------------------------------------------------------
+vital::config_block_sptr
+resection_camera
+::get_configuration() const
+{
+  // get base config from base class
+  vital::config_block_sptr config =
+    vital::algo::resection_camera::get_configuration();
+  config->set_value( "reproj_accuracy", d_->reproj_accuracy,
+                     "desired re-projection positive accuracy" );
+  config->set_value( "max_iterations", d_->max_iterations,
+                     "maximum number of iterations to run PnP [1, INT_MAX]" );
+  return config;
+}
+
+/// Set this algorithm's properties via a config block.
+void
+resection_camera
+::set_configuration( vital::config_block_sptr config )
+{
+  d_->reproj_accuracy = config->get_value< double >( "reproj_accuracy",
+                                                     d_->reproj_accuracy );
+  d_->max_iterations = config->get_value< int >( "max_iterations",
+                                                 d_->max_iterations );
+}
+
+/// Check that the algorithm's configuration vital::config_block is valid.
+bool
+resection_camera
+::check_configuration( vital::config_block_sptr config ) const
+{
+  bool good_conf = true;
+  double reproj_accuracy = config->get_value< double >( "reproj_accuracy",
+                                                        d_->reproj_accuracy );
+  if( reproj_accuracy <= 0.0 )
+  {
+    LOG_ERROR( d_->m_logger,
+               "reproj_accuracy parameter is " << reproj_accuracy <<
+               ", but needs to be positive." );
+    good_conf = false;
+  }
+
+  int max_iterations = config->get_value< int >( "max_iterations",
+                                                 d_->max_iterations );
+  if( max_iterations < 1 )
+  {
+    LOG_ERROR( d_->m_logger,
+               "max iterations is " << max_iterations <<
+               ", needs to be greater than zero." );
+    good_conf = false;
+  }
+  return good_conf;
+}
+
+vital::camera_perspective_sptr
+resection_camera
+::resection(
+  vector< vital::vector_2d > const& pts2d,
+  vector< vital::vector_3d > const& pts3d,
```

Comment:
Should we rename these for consistency with the names in the abstract algorithm?